src/Common/FitOnFhir.Common/Services/UsersServiceBase.cs

```diff
@@ -0,0 +1,49 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using EnsureThat;
+using FitOnFhir.Common.Models;
+using FitOnFhir.Common.Repositories;
+using Hl7.Fhir.Model;
+using Microsoft.Health.Extensions.Fhir.Service;
+
+namespace FitOnFhir.Common.Services
+{
+    public abstract class UsersServiceBase
+    {
+        private readonly ResourceManagementService _resourceManagementService;
+        private readonly IUsersTableRepository _usersTableRepository;
+
+        public UsersServiceBase(ResourceManagementService resourceManagementService, IUsersTableRepository usersTableRepository)
+        {
+            _resourceManagementService = EnsureArg.IsNotNull(resourceManagementService, nameof(resourceManagementService));
+            _usersTableRepository = EnsureArg.IsNotNull(usersTableRepository, nameof(usersTableRepository));
+        }
+
+        public async System.Threading.Tasks.Task EnsurePatientAndUser(string platformName, string identifier, string system, CancellationToken cancellationToken)
```

Comment:
Does this namespace need to be fully qualified?

Reply:
Unfortunately, yes. Hl7.Fhir.Model has a type named Task. It's frustrating.